operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java

```diff
@@ -533,6 +547,23 @@ private String onSeparateLines(List<String> lines) {
     }
   }
 
+  private static void logIntrospectorFailure(Packet packet, V1Job domainIntrospectorJob) {
+    Boolean logged = (Boolean) packet.get(ProcessingConstants.INTROSPECTOR_JOB_FAILURE_LOGGED);
+    String jobPodName = (String) packet.get(ProcessingConstants.JOB_POD_NAME);
+    if (logged == null || !logged.booleanValue()) {
+      packet.put(ProcessingConstants.INTROSPECTOR_JOB_FAILURE_LOGGED, Boolean.valueOf(true));
+      LOGGER.info(INTROSPECTOR_JOB_FAILED,
+          domainIntrospectorJob.getMetadata().getNamespace(),
+          domainIntrospectorJob.getMetadata().getName(),
+          jobPodName,
+          domainIntrospectorJob.getStatus());
+      LOGGER.fine(INTROSPECTOR_JOB_FAILED_DETAIL,
```

Comment:
could you give an example of what is being logged in this message? Thanks

Reply:
It is in the regular operator log message format with the message body be V1Job.toString().

Reply:
{"timestamp":"06-29-2020T18:17:16.125+0000","thread":20,"fiber":"engine-operator-thread-3-fiber-4","namespace":"sample-domain1-ns","domainUID":"sample-domain1","level":"FINE","class":"oracle.kubernetes.operator.helpers.JobHelper","method":"logIntrospectorFailure","timeInMillis":1593454636125,"message":"Job sample-domain1-introspect-domain-job in namespace sample-domain1-ns failed, job details are class V1Job {\n    apiVersion: batch/v1\n    kind: Job\n    metadata: class V1ObjectMeta {\n        annotations: null\n        clusterName: null\n        creationTimestamp: 2020-06-29T18:17:14.000Z\n        deletionGracePeriodSeconds: null\n        deletionTimestamp: null\n        finalizers: null\n        generateName: null\n        generation: null\n        labels: {weblogic.createdByOperator=true, weblogic.domainUID=sample-domain1}\n        managedFields: null\n        name: sample-domain1-introspect-domain-job\n        namespace: sample-domain1-ns\n        ownerReferences: [class V1OwnerReference {\n            apiVersion: weblogic.oracle/v8\n            blockOwnerDeletion: null\n            controller: true\n            kind: Domain\n            name: sample-domain1\n            uid: fb79795e-e997-4dd9-96b4-92a71f484188\n        }]\n        resourceVersion: 790406\n        selfLink: /apis/batch/v1/namespaces/sample-domain1-ns/jobs/sample-domain1-introspect-domain-job\n        uid: 3a051671-91aa-40de-b93d-64a692afd316\n    }\n    spec: class V1JobSpec {\n        activeDeadlineSeconds: 120\n        backoffLimit: 0\n        completions: 1\n        manualSelector: null\n        parallelism: 1\n        selector: class V1LabelSelector {\n            matchExpressions: null\n            matchLabels: {controller-uid=3a051671-91aa-40de-b93d-64a692afd316}\n        }\n        template: class V1PodTemplateSpec {\n            metadata: class V1ObjectMeta {\n                annotations: null\n                clusterName: null\n                creationTimestamp: null\n                deletionGracePeriodSeconds: null\n                deletionTimestamp: null\n                finalizers: null\n                generateName: null\n                generation: null\n                labels: {controller-uid=3a051671-91aa-40de-b93d-64a692afd316, job-name=sample-domain1-introspect-domain-job, weblogic.createdByOperator=true, weblogic.domainUID=sample-domain1}\n                managedFields: null\n                name: sample-domain1-introspect-domain-job\n                namespace: null\n                ownerReferences: null\n                resourceVersion: null\n                selfLink: null\n                uid: null\n            }\n            spec: class V1PodSpec {\n                activeDeadlineSeconds: 120\n                affinity: null\n                automountServiceAccountToken: null\n                containers: [class V1Container {\n                    args: null\n                    command: [/weblogic-operator/scripts/introspectDomain.sh]\n                    env: [class V1EnvVar {\n                        name: JAVA_OPTIONS\n                        value: -Dweblogic.StdoutDebugEnabled=false\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: USER_MEM_ARGS\n                        value: -Djava.security.egd=file:/dev/./urandom \n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: DOMAIN_UID\n                        value: sample-domain1\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: DOMAIN_HOME\n                        value: /u01/oracle/user_projects/domains\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: NODEMGR_HOME\n                        value: /u01/nodemanager\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: LOG_HOME\n                        value: null\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: SERVER_OUT_IN_POD_LOG\n                        value: true\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ACCESS_LOG_IN_LOG_HOME\n                        value: true\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: NAMESPACE\n                        value: sample-domain1-ns\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: INTROSPECT_HOME\n                        value: /u01/oracle/user_projects/domains\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: CREDENTIALS_SECRET_NAME\n                        value: sample-domain1-weblogic-credentials\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: OPSS_KEY_SECRET_NAME\n                        value: null\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: OPSS_WALLETFILE_SECRET_NAME\n                        value: null\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: RUNTIME_ENCRYPTION_SECRET_NAME\n                        value: null\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: WDT_DOMAIN_TYPE\n                        value: WLS\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: DOMAIN_SOURCE_TYPE\n                        value: Image\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ISTIO_ENABLED\n                        value: false\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ISTIO_READINESS_PORT\n                        value: 8888\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ISTIO_POD_NAMESPACE\n                        value: sample-domain1-ns\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: OPERATOR_ENVVAR_NAMES\n                        value: JAVA_OPTIONS,USER_MEM_ARGS,DOMAIN_UID,DOMAIN_HOME,NODEMGR_HOME,LOG_HOME,SERVER_OUT_IN_POD_LOG,ACCESS_LOG_IN_LOG_HOME,NAMESPACE,INTROSPECT_HOME,CREDENTIALS_SECRET_NAME,OPSS_KEY_SECRET_NAME,OPSS_WALLETFILE_SECRET_NAME,RUNTIME_ENCRYPTION_SECRET_NAME,WDT_DOMAIN_TYPE,DOMAIN_SOURCE_TYPE,ISTIO_ENABLED,ISTIO_READINESS_PORT,ISTIO_POD_NAMESPACE\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ADMIN_USERNAME\n                        value: null\n                        valueFrom: null\n                    }, class V1EnvVar {\n                        name: ADMIN_PASSWORD\n                        value: null\n                        valueFrom: null\n                    }]\n                    envFrom: null\n                    image: container-registry.oracle.com/middleware/weblogic:12.2.1.4\n                    imagePullPolicy: IfNotPresent\n                    lifecycle: null\n                    livenessProbe: null\n                    name: sample-domain1-introspect-domain-job\n                    ports: null\n                    readinessProbe: null\n                    resources: class V1ResourceRequirements {\n                        limits: null\n                        requests: null\n                    }\n                    securityContext: class V1SecurityContext {\n                        allowPrivilegeEscalation: null\n                        capabilities: null\n                        privileged: null\n                        procMount: null\n                        readOnlyRootFilesystem: null\n                        runAsGroup: null\n                        runAsNonRoot: null\n                        runAsUser: null\n                        seLinuxOptions: null\n                        windowsOptions: null\n                    }\n                    startupProbe: null\n                    stdin: null\n                    stdinOnce: null\n                    terminationMessagePath: /dev/termination-log\n                    terminationMessagePolicy: File\n                    tty: null\n                    volumeDevices: null\n                    volumeMounts: [class V1VolumeMount {\n                        mountPath: /weblogic-operator/secrets\n                        mountPropagation: null\n                        name: weblogic-credentials-volume\n                        readOnly: true\n                        subPath: null\n                        subPathExpr: null\n                    }, class V1VolumeMount {\n                        mountPath: /weblogic-operator/scripts\n                        mountPropagation: null\n                        name: weblogic-scripts-cm-volume\n                        readOnly: true\n                        subPath: null\n                        subPathExpr: null\n                    }, class V1VolumeMount {\n                        mountPath: /weblogic-operator/introspectormii\n                        mountPropagation: null\n                        name: mii-weblogic-domain-introspect-cm\n                        readOnly: null\n                        subPath: null\n                        subPathExpr: null\n                    }]\n                    workingDir: null\n                }]\n                dnsConfig: null\n                dnsPolicy: ClusterFirst\n                enableServiceLinks: null\n                ephemeralContainers: null\n                hostAliases: null\n                hostIPC: null\n                hostNetwork: null\n                hostPID: null\n                hostname: null\n                imagePullSecrets: null\n                initContainers: null\n                nodeName: null\n                nodeSelector: null\n                overhead: null\n                preemptionPolicy: null\n                priority: null\n                priorityClassName: null\n                readinessGates: null\n                restartPolicy: Never\n                runtimeClassName: null\n                schedulerName: default-scheduler\n                securityContext: class V1PodSecurityContext {\n                    fsGroup: null\n                    runAsGroup: null\n                    runAsNonRoot: null\n                    runAsUser: null\n                    seLinuxOptions: null\n                    supplementalGroups: null\n                    sysctls: null\n                    windowsOptions: null\n                }\n                serviceAccount: null\n                serviceAccountName: null\n                shareProcessNamespace: null\n                subdomain: null\n                terminationGracePeriodSeconds: 30\n                tolerations: null\n                topologySpreadConstraints: null\n                volumes: [class V1Volume {\n                    awsElasticBlockStore: null\n                    azureDisk: null\n                    azureFile: null\n                    cephfs: null\n                    cinder: null\n                    configMap: null\n                    csi: null\n                    downwardAPI: null\n                    emptyDir: null\n                    fc: null\n                    flexVolume: null\n                    flocker: null\n                    gcePersistentDisk: null\n                    gitRepo: null\n                    glusterfs: null\n                    hostPath: null\n                    iscsi: null\n                    name: weblogic-credentials-volume\n                    nfs: null\n                    persistentVolumeClaim: null\n                    photonPersistentDisk: null\n                    portworxVolume: null\n                    projected: null\n                    quobyte: null\n                    rbd: null\n                    scaleIO: null\n                    secret: class V1SecretVolumeSource {\n                        defaultMode: 420\n                        items: null\n                        optional: null\n                        secretName: sample-domain1-weblogic-credentials\n                    }\n                    storageos: null\n                    vsphereVolume: null\n                }, class V1Volume {\n                    awsElasticBlockStore: null\n                    azureDisk: null\n                    azureFile: null\n                    cephfs: null\n                    cinder: null\n                    configMap: class V1ConfigMapVolumeSource {\n                        defaultMode: 365\n                        items: null\n                        name: weblogic-scripts-cm\n                        optional: null\n                    }\n                    csi: null\n                    downwardAPI: null\n                    emptyDir: null\n                    fc: null\n                    flexVolume: null\n                    flocker: null\n                    gcePersistentDisk: null\n                    gitRepo: null\n                    glusterfs: null\n                    hostPath: null\n                    iscsi: null\n                    name: weblogic-scripts-cm-volume\n                    nfs: null\n                    persistentVolumeClaim: null\n                    photonPersistentDisk: null\n                    portworxVolume: null\n                    projected: null\n                    quobyte: null\n                    rbd: null\n                    scaleIO: null\n                    secret: null\n                    storageos: null\n                    vsphereVolume: null\n                }, class V1Volume {\n                    awsElasticBlockStore: null\n                    azureDisk: null\n                    azureFile: null\n                    cephfs: null\n                    cinder: null\n                    configMap: class V1ConfigMapVolumeSource {\n                        defaultMode: 365\n                        items: null\n                        name: sample-domain1-weblogic-domain-introspect-cm\n                        optional: true\n                    }\n                    csi: null\n                    downwardAPI: null\n                    emptyDir: null\n                    fc: null\n                    flexVolume: null\n                    flocker: null\n                    gcePersistentDisk: null\n                    gitRepo: null\n                    glusterfs: null\n                    hostPath: null\n                    iscsi: null\n                    name: mii-weblogic-domain-introspect-cm\n                    nfs: null\n                    persistentVolumeClaim: null\n                    photonPersistentDisk: null\n                    portworxVolume: null\n                    projected: null\n                    quobyte: null\n                    rbd: null\n                    scaleIO: null\n                    secret: null\n                    storageos: null\n                    vsphereVolume: null\n                }]\n            }\n        }\n        ttlSecondsAfterFinished: null\n    }\n    status: class V1JobStatus {\n        active: null\n        completionTime: null\n        conditions: [class V1JobCondition {\n            lastProbeTime: 2020-06-29T18:17:15.000Z\n            lastTransitionTime: 2020-06-29T18:17:15.000Z\n            message: Job has reached the specified backoff limit\n            reason: BackoffLimitExceeded\n            status: True\n            type: Failed\n        }]\n        failed: 1\n        startTime: 2020-06-29T18:17:14.000Z\n        succeeded: null\n    }\n}","exception":"","code":"","headers":{},"body":""}

Reply:
I tried to put an example here but the format got messed up by the tooling. I'll send you via slack.